File: frontend/src/types/asset.ts
export type AssetType =
  | 'VEHICULO'
  | 'ELECTRODOMESTICO'
  | 'INMUEBLE'
  | 'EQUIPO'
  | 'OTRO';

export const ASSET_TYPE_LABELS: Record<AssetType, string> = {
  VEHICULO: 'Vehículo',
  ELECTRODOMESTICO: 'Electrodoméstico',
  INMUEBLE: 'Inmueble',
  EQUIPO: 'Equipo',
  OTRO: 'Otro',
};

export interface Asset {
  id: number;
  customerId: number;
  tipo: AssetType;
  marca: string | null;
  modelo: string | null;
  placaSerial: string | null;
  anio: number | null;
  notas: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface AssetRequest {
  tipo: AssetType;
  marca?: string | null;
  modelo?: string | null;
  placaSerial?: string | null;
  anio?: number | null;
  notas?: string | null;
}

File: src/main/java/cotizaciones/app/Model/Asset.java
package cotizaciones.app.Model;



import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.OffsetDateTime;

@Entity
@Table(name = "assets")
@Getter
@Setter
public class Asset {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "customer_id", nullable = false)
    private Customer customer;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    private AssetType tipo;

    @Column(length = 60)
    private String marca;

    @Column(length = 60)
    private String modelo;

    @Column(name = "placa_serial", length = 60)
    private String placaSerial;

    @Column(name = "anio")
    private Integer anio;

    @Column(columnDefinition = "text")
    private String notas;

    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;

    @PrePersist
    void onCreate() {
        var now = OffsetDateTime.now();
        createdAt = now;
        updatedAt = now;
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = OffsetDateTime.now();
    }
}